jbpm-flow/src/main/java/org/jbpm/process/core/context/variable/VariableScope.java

```diff
@@ -91,22 +92,26 @@ public Context resolveContext(Object param) {
             "VariableScopes can only resolve variable names: " + param);
     }
     
-	public void validateVariable(String processName, String name, Object value) {
-		if (!variableStrictEnabled) {
-			return;
-		}
-		Variable var = findVariable(name);
-    	if (var == null) {
-    		throw new IllegalArgumentException("Variable '" + name +"' is not defined in process " + processName);
-    	}
-    	if (var.getType() != null && value != null) {
-	    	boolean isValidType = var.getType().verifyDataType(value);
-	    	if (!isValidType) {
-	    		throw new IllegalArgumentException("Variable '" + name +"' has incorrect data type expected:" 
-	    						+ var.getType().getStringType() + " actual:" + value.getClass().getName());
-	    	}
-    	}
-	}
+    public Object validateVariable(String processName, String name, Object value) {
+        Variable var = findVariable(name);
+        if (var == null) {
+            if (variableStrictEnabled) {
```

Comment:
you did change the semantics of the flag variableStrictEnabled

Reply:
no, if variableStrictEnabled is true, code behaves the same. I change behaviour when semantic is false, to try to convert values to the right data type and if that attemps fails, use the same object that was provided, so the only change is that if user pass a string that can be converted to int for a int data type, int will be used (as requested by customer)